### PR TITLE
Facebook Video downloader

### LINKED ISSUE #1037: Facebook video downloader (GUI)
# Aim

Download videos from Facebook

# Details

The user will pass the link of the video in the text field and you need to implement the logic to download the video. If possible, implement the progress bar to show the progress of the download. 

If progress bar, the level will be increased to 3 else 2.

### PR DESCRIPTION
**Pull request by Amit366**
# Description

Added script to download facebook video
## Type of change

_Please delete options that are not relevant._

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation Update

## Checklist:

- [ ] My code follows the style guidelines(Clean Code) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have created a helpful and easy to understand `README.md`
- [ ] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [ ] I have added the project meta data in the PR template.
- [ ] I have created the ``requirements.txt`` file if needed.

## Project Metadata

`` If there is no-file/nothing to fill the below fields with, then type: none ``

`` Example: `` If no requirements.txt needed/present, then type none in ``Requirments``.

Category:
- [ ] Calculators
- [ ] AI/ML
- [ ] Scrappers
- [ ] Social_Media
- [ ] PDF
- [ ] Image_Processing
- [ ] Video_Processing
- [ ] Games
- [ ] Networking
- [ ] OS_Utilities
- [ ] Automation
- [ ] Cryptography
- [ ] Computer_Vision
- [ ] Fun
- [ ] Others

Title: Fcaebook video downloader

Folder: Facebook_Video_Downloader
Requirements: urllib.requests

Script: script.py
Arguments: none

Contributor: Amit Kumar Mishra

Description: Added script to download facebook video

### CODE REVIEW

**Comment by Amit366** [comment]
@kaustubhgupta  can u please help me with progress bar?

File: Facebook_Video_downloader/script.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import re
+import requests
+import urllib.request
+from tqdm import tqdm
+
+video_link = input("Enter the video link: ")
+
+req = requests.get(video_link, stream=True)
+
+print("downloading......")
```
Comment (kaustubhgupta) [inline Facebook_Video_downloader/script.py:10]: This is not working, I tried with this link: https://www.facebook.com/herowarsgame/videos/961066781305631/

A file was downloaded but it's not playing. Also, as mentioned in the issue description, implement the GUI and then we will see if progress bar can be implemented

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by Amit366** [comment]
Ok

**Comment by kaustubhgupta** [comment]
@Amit366 updates on this?

**Comment by Amit366** [comment]
@kaustubhgupta i am unwell. So I request you to please give me 3 more days.

**Comment by kaustubhgupta** [comment]
> @kaustubhgupta i am unwell. So I request you to please give me 3 more days.

okay, take rest!

**Comment by Amit366** [comment]
@kaustubhgupta I tried implementing the progress bar but it is showing an error .

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Not working (link tried:  https://www.facebook.com/cistheta/videos/983590405510302)

image

Also, you are again doing plagiarism: https://github.com/avinashkranjan/Amazing-Python-Scripts/pull/1046/files. You have copied the code from this PR as it is without even changing fonts, comments, displayed a big blue screen, the purpose of that is still unpredictable. 

I can easily put 3rd strike on this which would nullify all your scores in entire GSSOC. But I am giving you last chance to improve. Come up with GUI of your own, fix the video downloading issues.

**Comment by Amit366** [comment]
@kaustubhgupta I can do it with progress bar without GUI

**Comment by Amit366** [comment]
If you want I can do else I cannot do GUI

**Comment by kaustubhgupta** [comment]
Implement it with GUI and without progress bar

**Comment by Amit366** [comment]
@kaustubhgupta I am facing problem with GUI that's why I am asking you that can I implement without GUI

**Comment by kaustubhgupta** [comment]
> @kaustubhgupta I am facing problem with GUI that's why I am asking you that can I implement without GUI

this issue is supposed to be solved in GUI only. If you're facing issues then feel free to drop it. Also, as you are facing issues with this, I think you won't be able to solve the LinkedIn video downloader GUI too.

**Pull request closed without merging**